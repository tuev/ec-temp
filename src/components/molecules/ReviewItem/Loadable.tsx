/**
 *
 * Asynchronously loads the component for ReviewItem
 *
 */

import loadable from 'utils/loadable'

export default loadable(() => import('./index'))
