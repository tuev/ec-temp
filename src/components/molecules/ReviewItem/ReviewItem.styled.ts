/**
 *
 * Styled for ReviewItem
 *
 */

import styled from 'styled-components'
import { Typography } from 'components/atoms'
import { COLOR } from 'theme/colors'

export const Warning = styled(Typography)`
  color: ${COLOR.black2};
`
