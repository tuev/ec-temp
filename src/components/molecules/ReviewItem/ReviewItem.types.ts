export interface IReviewItem {
  id: number
  rating: number
  fullname: string
  dateCreated: string
  reviewText: string
  likeCount: number
  commentCount: number
}
