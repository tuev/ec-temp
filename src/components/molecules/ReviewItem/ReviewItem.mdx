# Stories mdx for ReviewItem

import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './ReviewItem.stories.tsx'
import ReviewItem from './ReviewItem.tsx'

<Story name="basic" />