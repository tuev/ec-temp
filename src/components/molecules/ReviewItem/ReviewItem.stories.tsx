/**
 *
 *  Stories for ReviewItem
 *
 */

import React from 'react'
import { action } from '@storybook/addon-actions'
import ReviewItem from './ReviewItem'

export default {
  component: ReviewItem,
  includeStories: ['basic'],
  title: 'ReviewItem',
}

export const basic = () => (
  <div>
    <ReviewItem />
  </div>
)
