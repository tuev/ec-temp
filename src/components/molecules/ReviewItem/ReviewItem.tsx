/**
 *
 * ReviewItem
 *
 */

import React, { FC } from 'react'

import { Text } from '../ReviewForm/ReviewForm.styled'
import { Grid } from '@material-ui/core'
import { Rating, Typography, BlogIcon } from 'components/atoms'
import { Warning } from './ReviewItem.styled'
import { IReviewItem } from './ReviewItem.types'

const ReviewItem: FC<IReviewItem> = ({
  rating,
  fullname,
  dateCreated,
  reviewText,
  likeCount,
  commentCount,
}) => {
  return (
    <>
      <Grid item={true} xs={12} md={3}>
        <Rating value={rating} readOnly={true} mb={4} />
        <Text customvariant="body1">{fullname}</Text>
        <Warning customvariant="body2">{dateCreated}</Warning>
      </Grid>
      <Grid item={true} xs={12} md={9}>
        <Typography customvariant="body1" mb={3}>
          {reviewText}
        </Typography>
        <Grid container={true}>
          <Grid item={true} xs={2}>
            <BlogIcon icon="like" count={likeCount} />
          </Grid>
          <Grid item={true} xs={2}>
            <BlogIcon icon="comment" count={commentCount} />
          </Grid>
        </Grid>
      </Grid>
    </>
  )
}

export default ReviewItem
