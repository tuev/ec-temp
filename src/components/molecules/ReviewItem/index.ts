/**
 *
 * Index for ReviewItem
 *
 */

import ReviewItem from './ReviewItem'

export default ReviewItem
