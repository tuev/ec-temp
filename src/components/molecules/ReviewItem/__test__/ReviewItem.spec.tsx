/**
 *
 * Tests for ReviewItem
 *
 */

import React from 'react'

import { render, fireEvent } from '@testing-library/react'
import ReviewItem from '..'

describe('App ReviewItem', () => {
  it('Test get default ReviewItem no params ', () => {
    // test snapshot
    const wrapper = render(<ReviewItem />)
    expect(wrapper).toMatchSnapshot()
  })
})
