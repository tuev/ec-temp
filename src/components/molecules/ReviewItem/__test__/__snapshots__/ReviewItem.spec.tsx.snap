// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App ReviewItem Test get default ReviewItem no params  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  font-size: 16px;
  line-height: 1.5;
  font-family: 'Open Sans';
  color: #2E2E2E;
}

.c3 {
  font-size: 12px;
  line-height: 2;
  font-family: 'Open Sans';
  color: #2E2E2E;
}

.c5 {
  margin-bottom: 16px;
  font-size: 16px;
  line-height: 1.5;
  font-family: 'Open Sans';
  color: #2E2E2E;
}

.c2 {
  font-weight: 600;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  max-width: 80px;
  cursor: pointer;
}

.c7 {
  width: 32px;
  height: 32px;
  border-radius: 50%;
  background-color: #F2F2F2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  margin-bottom: 32px;
}

.c4 {
  color: #B4B4B4;
}

<body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3"
      >
        <span
          aria-label="null Stars"
          class="MuiRating-root c0 MuiRating-readOnly"
          mb="4"
          role="img"
        >
          <span>
            <span
              class="MuiRating-icon MuiRating-iconEmpty"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconEmpty"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconEmpty"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconEmpty"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconEmpty"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <p
          class="MuiTypography-root c1 c2 MuiTypography-body1"
          customvariant="body1"
        />
        <p
          class="MuiTypography-root c3 c4 MuiTypography-body1"
          customvariant="body2"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-9"
      >
        <p
          class="MuiTypography-root c5 MuiTypography-body1"
          customvariant="body1"
          mb="3"
        />
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
          >
            <div
              class="c6"
            >
              <div
                class="c7"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
                  />
                </svg>
              </div>
              <span />
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
          >
            <div
              class="c6"
            >
              <div
                class="c7"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H6l-2 2V4h16v12z"
                  />
                </svg>
              </div>
              <span />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3"
    >
      <span
        aria-label="null Stars"
        class="MuiRating-root sc-fzozJi fnALdj MuiRating-readOnly"
        mb="4"
        role="img"
      >
        <span>
          <span
            class="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
          </span>
        </span>
      </span>
      <p
        class="MuiTypography-root sc-AxjAm FxMPz sc-AxiKw cvnEaT MuiTypography-body1"
        customvariant="body1"
      />
      <p
        class="MuiTypography-root sc-AxjAm dATcSb sc-fzpans kCLvXU MuiTypography-body1"
        customvariant="body2"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-9"
    >
      <p
        class="MuiTypography-root sc-AxjAm gyLNiV MuiTypography-body1"
        customvariant="body1"
        mb="3"
      />
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
        >
          <div
            class="sc-AxhUy gWAFto"
          >
            <div
              class="sc-AxgMl cTuaLz"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
                />
              </svg>
            </div>
            <span />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
        >
          <div
            class="sc-AxhUy gWAFto"
          >
            <div
              class="sc-AxgMl cTuaLz"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H6l-2 2V4h16v12z"
                />
              </svg>
            </div>
            <span />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
